### PR TITLE
User can login with spotify

### PR DESCRIPTION
**Pull request by sealfury**
[API: User must log in with Spotify account to use app](https://www.pivotaltracker.com/story/show/176623888)
### User Story
```
As a an API,            
In order to uphold the terms of Spotify services,         
I want to ensure that users are authenticated through Spotify.
```
## Changes proposed in this pull request:
* adds OmniAuth login configuration 
* configures `devise_token_auth` to use `omniauth` to enable login to app and spotify simultaneously 

## Packages/Gems added
* `gem "omniauth", "~> 1.9.1"` : **Obs: version must not be upgraded beyond `1.9.1`**
* `gem 'devise_token_auth'`
* `gem 'omniauth-spotify'`
* `gem 'guard-rspec'`

### CODE REVIEW

File: app/controllers/api/omniauth_callbacks_controller.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    protected
+
+    def auth_hash
+      request.env['omniauth.auth']
+    end
+
+    def default_devise_mapping
+      'User'.constantize
+    end
+
```
Comment (salindholm) [inline app/controllers/api/omniauth_callbacks_controller.rb:33]: Please remove this extra line

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+class User < ActiveRecord::Base
+  extend Devise::Models
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :trackable, :validatable,
+         :omniauthable, omniauth_providers: [:spotify]
+  include DeviseTokenAuth::Concerns::User
+
```
Comment (salindholm) [inline app/models/user.rb:8]: Please remove this extra line

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+
```
Comment (salindholm) [inline spec/models/user_spec.rb]: Please remove this extra line

File: spec/rails_helper.rb [hunk 4]
```diff
@@ -9,6 +9,8 @@
 require 'rspec/rails'
 require 'spec_helper'
 require 'webmock/rspec'
+require 'rspotify'
+
```
Comment (salindholm) [inline spec/rails_helper.rb:13]: Please remove this extra line

**Review by salindholm — CHANGES_REQUESTED** [review]
Good job, left some comments about extra lines! Also make sure to fix Semaphore

File: spec/rails_helper.rb (hunk 4 above)
Comment (ViktorHek) [reply spec/rails_helper.rb:13]: done :)

File: app/models/user.rb (hunk 2 above)
Comment (ViktorHek) [reply app/models/user.rb:8]: done :)

File: spec/models/user_spec.rb (hunk 3 above)
Comment (ViktorHek) [reply spec/models/user_spec.rb]: done :)

File: app/controllers/api/omniauth_callbacks_controller.rb (hunk 1 above)
Comment (ViktorHek) [reply app/controllers/api/omniauth_callbacks_controller.rb:33]: done :)

**Pull request merged**